extensions/wikia/PortableInfobox/services/Parser/Nodes/Node.php

```diff
@@ -83,10 +94,51 @@ public function getData() {
 	 *
 	 * @return bool
 	 */
-	public function isEmpty( $data ) {
-		$value = $data[ 'value' ];
+	public function isEmpty() {
+		$data = $this->getData()[ 'value' ];
+
+		return !( isset( $data ) ) || ( empty( $data ) && $data != '0' );
+	}
+
+	protected function getChildNodes() {
+		if ( !isset( $this->children ) ) {
+			$this->children = [ ];
+			foreach ( $this->xmlNode as $child ) {
+				$this->children[ ] = NodeFactory::newFromSimpleXml( $child, $this->infoboxData )
+					->setExternalParser( $this->externalParser );
+			}
+		}
 
-		return !( isset( $value ) ) || ( empty( $value ) && $value != '0' );
+		return $this->children;
+	}
+
+	protected function getDataForChildren() {
+		return array_map( function ( Node $item ) {
+			return [
+				'type' => $item->getType(),
+				'data' => $item->getData(),
+				'isEmpty' => $item->isEmpty(),
+				'source' => $item->getSource()
+			];
+		}, $this->getChildNodes() );
+	}
+
+	protected function getRenderDataForChildren() {
```

Comment:
This reads a bit hard because of formatting and using anonymous functions. 
Can we reformat so each param of array_map starts with the same level of indentation?